pkg/controller/tektoninstallation/tektoninstallation_controller.go

```diff
@@ -87,41 +107,129 @@ func (r *ReconcileTektonInstallation) Reconcile(request reconcile.Request) (reco
 		return reconcile.Result{}, err
 	}
 
-	err := r.EnsureTektonSubscription(reqLogger, tektonInstallation)
-	return reconcile.Result{}, err
+	if created, err := r.EnsureTektonSubscription(reqLogger, tektonInstallation); err != nil {
+		return reconcile.Result{}, err
+	} else if created {
+		return reconcile.Result{}, r.statusUpdate(reqLogger, tektonInstallation, r.setStatusTektonSubscriptionCreated, "created tekton subscription")
+	}
+
+	if requeue, err := r.ensureWatchTektonConfig(); err != nil {
+		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, tektonInstallation, r.setStatusTektonInstallationFailed, err, "failed to add watch for TektonConfig")
+	} else if requeue {
+		return reconcile.Result{Requeue: true, RequeueAfter: 3 * time.Second}, nil
+	}
+
+	cluster := &config.Config{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: TektonConfigName}, cluster)
+	if err != nil {
+		reqLogger.Info("requeue to retrieve cluster resource", "tektonconfigname", TektonConfigName)
+		return reconcile.Result{Requeue: true, RequeueAfter: 3 * time.Second}, nil
```

Comment:
Why requeueing? 
```suggestion
		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, tektonInstallation, r.setStatusTektonInstallationFailed, err, "failed to add get TektonConfig")
```

Reply:
@alexeykazakov i thought to requeue in order to get the config so we could appropriately set the status in the switch block below. i guess you are suggesting if we don't get it, we can fall into the default block, no need to requeue? is that correct?